### PR TITLE
improve error handling

### PR DESCRIPTION
**Pull request by xh3b4sd**
This is to have a more straight forward code flow and more consistent error handling.

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -224,6 +223,7 @@ func (c *Controller) Reconcile(req reconcile.Request) (reconcile.Result, error)
 
 	res, err := c.reconcile(ctx, req)
 	if err != nil {
+		c.errorCollector <- err
```
Comment (xh3b4sd) [inline controller/controller.go:226]: We should now get all errors collected without much hustle.

File: controller/controller.go [hunk 2]
```diff
@@ -329,50 +329,33 @@ func (c *Controller) bootWithError(ctx context.Context) error {
...
-
-		ctx, err = c.initCtx(ctx, obj)
-		if err != nil {
-			c.logger.LogCtx(oldCtx, "level", "error", "message", "failed initializing context", "stack", microerror.JSON(err))
-			c.logger.LogCtx(oldCtx, "level", "debug", "message", "canceling reconciliation")
-			return
-		}
+	ctx, err = c.initCtx(ctx, obj)
+	if err != nil {
+		return microerror.Mask(err)
```
Comment (xh3b4sd) [inline controller/controller.go]: This is now all way simpler.

File: controller/controller.go [hunk 3]
```diff
@@ -406,9 +389,10 @@ func (c *Controller) reconcile(ctx context.Context, req reconcile.Request) (reco
 		ctx = setLoggerCtxValue(ctx, loggerKeyObject, m.GetSelfLink())
 		ctx = setLoggerCtxValue(ctx, loggerKeyVersion, m.GetResourceVersion())
 
-		c.logger.LogCtx(ctx, "level", "debug", "message", "reconciling object")
-		c.deleteFunc(ctx, obj)
-		c.logger.LogCtx(ctx, "level", "debug", "message", "reconciled object")
```
Comment (xh3b4sd) [inline controller/controller.go:411]: I dropped a couple of logs that I thought became irrelevant over time. This was more noise than signal.

File: controller/controller.go [hunk 4]
```diff
@@ -390,6 +368,11 @@ func (c *Controller) reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, microerror.Mask(err)
 	}
 
+	ctx, err = c.initCtx(ctx, obj)
+	if err != nil {
+		return reconcile.Result{}, microerror.Mask(err)
+	}
```
Comment (xh3b4sd) [inline controller/controller.go:374]: I moved this up so it happens earlier and only in one place.

**Comment by kopiczko** [comment]
I wonder if we even need this collector now.

**Comment by xh3b4sd** [comment]
We use the metrics for the `OperatorkitErrorRateTooHigh*` alerts. Do you think we do not need them?

**Comment by kopiczko** [comment]
I think we can call `errorGauge.Inc()` directly instead of sending the error to the channel. See https://github.com/giantswarm/operatorkit/pull/403 (I didn't mean to isnta-merge it but GitHub).

**Pull request merged**